src/osd/osd_types.cc

```diff
@@ -5096,20 +5096,6 @@ void ScrubMap::decode(bufferlist::iterator& bl, int64_t pool)
   ::decode(valid_through, bl);
   ::decode(incr_since, bl);
   DECODE_FINISH(bl);
-
-  // handle hobject_t upgrade
-  if (struct_v < 3) {
-    map<hobject_t, object, hobject_t::BitwiseComparator> tmp;
-    tmp.swap(objects);
```

Comment:
You can take a look at the diff between them.  I found another bug with the range_calculation in chunky_scrub other that the one you fixed.  The new patch I added to fix the snap_mapper error necessitates restarting testing anyway -- it's a really really rare race, and we need to run a large set of tests to validate it.